--- a/espeak-phonemizer-1.0.0/src/lib.rs
+++ b/espeak-phonemizer-1.0.0/src/lib.rs
@@ -1,5 +1,5 @@
-#![feature(ptr_sub_ptr)]
-#![feature(str_lines_remainder)]
+
+
 
 mod espeakng;
 
@@ -58,8 +58,8 @@
     let mut phonemes = Vec::new();
     let total_length = text.len();
     let mut offset = 0;
-    let mut lines = text.lines();
-    while let Some(line) = lines.next() {
+    for line_with_sep in text.split_inclusive('\n') {
+        let line = line_with_sep.trim_end_matches('\n').trim_end_matches('\r');
         phonemes.append(&mut _text_to_phonemes(
             line,
             offset,
@@ -69,7 +69,7 @@
             remove_lang_switch_flags,
             remove_stress,
         )?);
-        offset = total_length - lines.remainder().unwrap_or_default().len()
+        offset += line_with_sep.len();
     }
     Ok(phonemes)
 }
@@ -147,7 +147,7 @@
             end = if !text_c_char.is_null() {
                 unsafe {
                     // text_c_char = text_c_char.sub(1);
-                    text_c_char.sub_ptr(text_c_char_prev) + start
+                    text_c_char.offset_from(text_c_char_prev) as usize + start
                 }
             } else {
                 text.len()
